src/org/stringtemplate/v4/Interpreter.java

```diff
@@ -495,29 +494,31 @@ void super_new(InstanceScope scope, String name, int nargs) {
             st = self.groupThatCreatedThisInstance.createStringTemplateInternally(new CompiledST());
         }
         else {
-            st = imported.nativeGroup.getEmbeddedInstanceOf(this, scope, name);
+            // TODO as this method was extracted from the two super_new overloads,
+            //      there was this difference in which method was called,
+            //      hence the addition of the 'embedded' parameter.
+            //      One of the calls is probably wrong.
+            if (embedded) {
+                st = imported.nativeGroup.getEmbeddedInstanceOf(this, scope, name);
+            }
+            else {
+                st = imported.nativeGroup.createStringTemplateInternally(imported);
+            }
```

Comment:
This needs some review. I'm not sure whether this difference was intended, perhaps someone changed one method and forgot the other.